### PR TITLE
fix: update notification count types to use number instead of string

### PR DESCRIPTION
**Pull request by Deshan1100**
### Summary

-

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

Clean refactor that tightens NotificationBadge's count prop to `number` and adds a '99+' overflow display via a new constant. Type changes are propagated correctly through Drawer.tsx and the four call sites in getDrawerRoutes.ts, and the sole consumer of NotificationBadge supplies numeric values from a typed hook. Minor concerns are around the constant's name and a small consistency point about where the overflow logic lives.

Found **11** new issue(s): 🔴 1 important, 🟡 3 suggestion(s), 🟣 7 nit(s)

File: frontend/src/community/common/constants/commonConstants.ts [hunk 1]
```diff
@@ -32,3 +32,4 @@ export const EVENTS_TO_IDENTIFY_IDLE_USER = [
 
 export const MODAL_HEADER_ICON_GAP = "3.5rem";
 export const DEFAULT_SCROLL_AMOUNT_PX = 100;
+export const MAX_NOTIFICATION_COUNT = 99;
```
Comment (ThinuwanW) [inline frontend/src/community/common/constants/commonConstants.ts:35]: 🤖 **Claude** · 🟣 **Nit**

The constant name `MAX_NOTIFICATION_COUNT` is slightly misleading — it is not the maximum count value possible, but the threshold above which the badge collapses to `99+`. A name like `NOTIFICATION_BADGE_DISPLAY_THRESHOLD` (or `MAX_DISPLAYED_NOTIFICATION_COUNT`) more accurately describes its role and avoids readers assuming 99 is a hard cap on the underlying counter. Optional rename for clarity.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx [hunk 2]
```diff
@@ -14,6 +16,9 @@ const NotificationBadge = ({
 
   if (!show) return null;
 
+  const displayCount =
+    count > MAX_NOTIFICATION_COUNT ? `${MAX_NOTIFICATION_COUNT}+` : count;
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: 🤖 **Claude** · 🟣 **Nit**

`displayCount` has the inferred type `string | number`. That is fine for React rendering, but if you ever need to format/localize the number later (e.g., `toLocaleString()`), you'll have to branch on the type. Consider always converting to string here (`count > MAX ? \`${MAX}+\` : String(count)`) so the variable is uniformly typed. Optional.

Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: 🤖 **Claude** · 🟡 **Suggestion**

With `count` now strictly `number`, a value of `0` (or a negative value from a future bug) would render an empty/zero badge if a future caller forgets the truthy guard. Today both call sites in Drawer.tsx gate on `route?.notificationCount &&`, so this is latent. Consider adding `if (count <= 0) return null;` inside the component so it is self-defensive and the guard does not need to be duplicated at every caller. Optional but reduces foot-gun risk now that the prop is strictly numeric.

File: frontend/src/community/common/components/organisms/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -392,7 +392,7 @@ const Drawer = (): JSX.Element => {
                           if (!subTreeRoute) return null;
                           const subRoute =
                             subTreeRoute as typeof subTreeRoute & {
-                              notificationCount?: string;
+                              notificationCount?: number;
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/organisms/Drawer/Drawer.tsx:395]: 🤖 **Claude** · 🟣 **Nit**

This inline `as typeof subTreeRoute & { notificationCount?: number; }` cast is a workaround because the upstream `subTree` type (from `routes` data) doesn't declare `notificationCount`. With this PR you're already touching the type, so it would be a good moment to surface `notificationCount?: number` on the actual return type of `getDrawerRoutes` (or on the subRoute object type) instead of relying on an `as` cast at the consumer. Out of scope to do here if you'd rather keep the PR focused, but worth a follow-up — the cast silently masks any future shape drift.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx [hunk 4]
```diff
@@ -14,6 +16,9 @@ const NotificationBadge = ({
...
+
   return (
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:22]: 🤖 **Claude** · 🔴 **Important**

Accessibility gap: the badge renders only a visual number ('5', '99+') with no semantic context for assistive tech. Screen-reader users will hear a bare digit next to the menu label with no indication of what it means. Add an aria-label like `aria-label={`${count} pending notifications`}` (ideally i18n-keyed per route — leave/timesheet/sign), and/or set `role="status"`/`aria-live="polite"` so updates are announced. Currently the parent `ListItemText` also doesn't reference the count via aria-describedby, so the badge is effectively invisible to screen readers.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx (hunk 2 above)
Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: 🤖 **Claude** · 🟣 **Nit**

The `+` suffix and ordering are hardcoded inline. For locales that mark a numeric ceiling differently (RTL formatting, or numerals via Intl.NumberFormat), this won't adapt. Consider sourcing the suffix through the existing translator (`useTranslator`) so the overflow indicator can be localized — e.g. `translateText(['notificationOverflow'], { max: MAX_NOTIFICATION_COUNT })`.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import { Box, useTheme } from "@mui/material";
 import { JSX } from "react";
 
+import { MAX_NOTIFICATION_COUNT } from "~community/common/constants/commonConstants";
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:4]: 🤖 **Claude** · 🟡 **Suggestion**

📁 **File-level observation**

No unit tests were added for the new threshold branch. The boundary behavior (count = MAX, count = MAX+1, count = 0, negative, NaN, very large) is exactly the kind of branch that quietly regresses on a future refactor. Add a small RTL/Jest test that asserts: renders the literal count at and below MAX, renders `${MAX}+` above MAX, and returns null when `show=false`.

File: frontend/src/community/common/constants/commonConstants.ts (hunk 1 above)
Comment (ThinuwanW) [inline frontend/src/community/common/constants/commonConstants.ts:35]: 🤖 **Claude** · 🟣 **Nit**

`MAX_NOTIFICATION_COUNT` is exported from the project-wide commonConstants module but has exactly one consumer (NotificationBadge). Co-locating it next to NotificationBadge.tsx (e.g. a sibling `constants.ts` or top-of-file constant) keeps the value with the only component that knows its semantics — and avoids growing commonConstants.ts into a junk drawer. Either move it, or rename to something less generic (e.g. `NOTIFICATION_BADGE_OVERFLOW_THRESHOLD`) so its meaning is obvious to a reader scrolling commonConstants.

File: frontend/src/community/common/utils/getDrawerRoutes.ts [hunk 6]
```diff
@@ -314,23 +312,23 @@ const getDrawerRoutes = ({
             if (subRoute.id === "2B" && notificationLeaveCount > 0) {
               return {
                 ...subRoute,
-                notificationCount: notificationLeaveCount.toString()
+                notificationCount: notificationLeaveCount
```
Comment (ThinuwanW) [inline frontend/src/community/common/utils/getDrawerRoutes.ts:315]: 🤖 **Claude** · 🟡 **Suggestion**

The three blocks at lines 312–333 (`2B`/leave, `1B`/timesheet, `4A`/sign) are now identical except for the id and count — the diff makes the duplication especially visible after removing `.toString()`. Consider replacing them with a small map (e.g. `const NOTIFICATION_COUNT_BY_SUB_ID: Record<string, number> = { '2B': notificationLeaveCount, '1B': notificationTimesheetCount, '4A': notificationSignCount }` and a single conditional adding `notificationCount` if `> 0`). Reduces the chance of future drift when a new subroute is added.

File: frontend/src/community/common/utils/getDrawerRoutes.ts [hunk 7]
```diff
@@ -294,9 +294,7 @@ const getDrawerRoutes = ({
             icon: route?.icon,
             hasSubTree: false,
             notificationCount:
-              notificationSignCount > 0
-                ? notificationSignCount.toString()
-                : undefined
+              notificationSignCount > 0 ? notificationSignCount : undefined
```
Comment (ThinuwanW) [inline frontend/src/community/common/utils/getDrawerRoutes.ts:297]: 🤖 **Claude** · 🟣 **Nit**

Inconsistency vs. the subroute branches: the Sign route here explicitly sets `notificationCount: undefined` when zero, while the subroute branches (lines 312/320/328) simply omit the field. Functionally equivalent but stylistically inconsistent — pick one form. Simpler is to drop the ternary and only conditionally spread: `...(notificationSignCount > 0 && { notificationCount: notificationSignCount })`.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx [hunk 8]
```diff
@@ -14,6 +16,9 @@ const NotificationBadge = ({
 
   if (!show) return null;
 
+  const displayCount =
```
Comment (ThinuwanW) [inline frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:19]: 🤖 **Claude** · 🟣 **Nit**

`count: number` doesn't constrain to non-negative integers — TypeScript `number` permits NaN, Infinity, and floats. The current logic would render 'NaN' (NaN > 99 is false), 'Infinity' (true → '99+'), or '50.5' if the API ever returned non-integer values. Either coerce defensively (`Math.max(0, Math.floor(count))`) before comparison, or document the contract with a comment and rely on callers — but the silent NaN render in particular is worth guarding.

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx (hunk 5 above)
Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:4]: No need a test

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx (hunk 8 above)
Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:19]: no negative numbers for pending notification count

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx (hunk 2 above)
Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: no any localize numbers

Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: if 0 its not display

Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:20]: thats fine only added + for count

File: frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx (hunk 4 above)
Comment (Deshan1100) [reply frontend/src/community/common/components/atoms/NotificationBadge/NotificationBadge.tsx:22]: kept as previous

File: frontend/src/community/common/components/organisms/Drawer/Drawer.tsx (hunk 3 above)
Comment (Deshan1100) [reply frontend/src/community/common/components/organisms/Drawer/Drawer.tsx:395]: always notifications getting as number that type safe

File: frontend/src/community/common/utils/getDrawerRoutes.ts (hunk 6 above)
Comment (Deshan1100) [reply frontend/src/community/common/utils/getDrawerRoutes.ts:315]: no need to convert to string can use num as it is

File: frontend/src/community/common/utils/getDrawerRoutes.ts (hunk 7 above)
Comment (Deshan1100) [reply frontend/src/community/common/utils/getDrawerRoutes.ts:297]: if notification count 0 noo need to pass as value therefore undefined

File: frontend/src/community/common/constants/commonConstants.ts (hunk 1 above)
Comment (Deshan1100) [reply frontend/src/community/common/constants/commonConstants.ts:35]: MAX_NOTIFICATION_COUNT thats fine

Comment (Deshan1100) [reply frontend/src/community/common/constants/commonConstants.ts:35]: MAX_NOTIFICATION_COUNT thats fine

**Review by shakila-kamalasena — APPROVED** [review]

**Review by hasalarootcode — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
